### PR TITLE
Add example of testing a dexterity type using a behavior. fix #229

### LINKED ISSUE #229: Document plone behavior tests for dexterity types.
To enable a behavior it is necessary to invalidate the Schema, otherwise it is not visible for the tests.

It is possible to see an example here: https://github.com/collective/collective.cover/blob/master/src/collective/cover/tests/test_refresh_behavior.py#L20

And here: https://github.com/ramiroluz/interlegis.portalmodelo.ombudsman/blob/a0e87710946ba91d62806730b4f185e9cdf7b7f8/src/interlegis/portalmodelo/ombudsman/tests/test_claims_by_tag.py#L50

### PR DESCRIPTION
**Pull request by ramiroluz**
I believe it works in Plone 5 and Plone 4.

### CODE REVIEW

**Comment by jensens** [comment]
Thanks, awesome!

File: docs/behaviors/testing-behaviors.rst [hunk 1]
```diff
@@ -173,6 +173,74 @@ To run the test, we can use the usual test runner:
 
     $ ./bin/instance test -s collective.gtags
 
+Testing a dexterity type with a behavior
+----------------------------------------
+
+Lets say you want to test your dexterity type when a behavior is enabled or disabled, note that not all behaviors are enabled by default. To do this you will need to setup the behavior in your test. There is an example of this kind of test in the collective.cover product. There is a behavior that adds the capability for the cover page to refresh itself. The test check to see if the behavior effect on the page is not present, then enables the behavior, check its effect and then disables it again. Here is the code:
```
Comment (jensens) [inline docs/behaviors/testing-behaviors.rst]: May you break the line after each sentence? 
This makes a translators life easier and is also what our documentation style guide recommends.

File: docs/behaviors/testing-behaviors.rst [hunk 2]
```diff
@@ -173,6 +173,74 @@ To run the test, we can use the usual test runner:
...
+            self.cover.enable_refresh = True
+            self.assertIn('<meta http-equiv="refresh" content="300" />', view())
+            self.cover.ttl = 5
+            self.assertIn('<meta http-equiv="refresh" content="5" />', view())
+            self._disable_refresh_behavior()
+            self.assertNotIn('<meta http-equiv="refresh" content="5" />', view())
+
+
+The methods ``_enable_refresh_behavior`` and ``_disable_refresh_behavior`` are using the ``IDexterityFTI``, to get the Factory Type Information for the dexterity type, in this case ``collective.cover.content``. Then the FTI of ``collective.cover.content`` is used by both methods to get a list of behaviors enabled. To enable it the desired behavior it is added to the FTI behaviors, ``behaviors.append(IRefresh.__identifier__)``. To disable it the behavior is removed from the FTI behaviors, ``behaviors.remove(IRefresh.__identifier__)``. The resulting behaviors list is assigned to the behaviors attribute of the FTI as a tuple, ``fti.behaviors = tuple(behaviors)``. Finally, to make the changes effective, the schema cache must be invalidated, ``notify(SchemaInvalidatedEvent('collective.cover.content'))``.
+
```
Comment (jensens) [inline docs/behaviors/testing-behaviors.rst]: same as above.

Comment (ramiroluz) [inline docs/behaviors/testing-behaviors.rst]: Ok, just a minute.

**Comment by tkimnguyen** [comment]
Needs just a bit of wording tweaks. I'll do that in the next 2 hours

**Comment by jensens** [comment]
Nice, that's how documentation should work always, thanks @ramiroluz and @tkimnguyen

**Pull request merged**